staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -472,80 +478,65 @@ type processorListener struct {
 
 func newProcessListener(handler ResourceEventHandler, requestedResyncPeriod, resyncPeriod time.Duration, now time.Time) *processorListener {
 	ret := &processorListener{
-		pendingNotifications:  []interface{}{},
 		nextCh:                make(chan interface{}),
+		addCh:                 make(chan interface{}),
```

Comment:
Do we really want this unbuffered? I noticed that in the case of just creating the processListener, we created a sync point because we need to hand off the notification to the newly created listener.